apps/user-office-frontend/src/components/proposal/ParticipantModal.tsx

```diff
@@ -30,61 +34,92 @@ function ParticipantModal(props: {
   invitationUserRole?: UserRole;
   participant?: boolean;
-}) {
+};
+
+const ParticipantModal: React.FC<ParticipantModalProps> = ({
+  addParticipants,
+  close,
+  show,
+  title,
+  invitationUserRole,
+  participant,
+  selectedUsers,
+  selection,
+  userRole,
+}) => {
+  const classes = useStyles();
+  const [selectedParticipants, setSelectedParticipants] = useState<
+    BasicUserDetails[]
+  >([]);
+  const isUserOfficer = useCheckAccess([UserRole.USER_OFFICER]);
+
   const addUser = (rowData: BasicUserDetails | BasicUserDetails[]) => {
     const addedUserDetails = rowData as BasicUserDetails;
 
-    props.addParticipants([addedUserDetails]);
+    addParticipants([addedUserDetails]);
+  };
+
+  const onClickHandlerUpdateBtn = () => {
+    addParticipants(selectedParticipants);
+    setSelectedParticipants([]);
   };
-  const classes = useStyles();
 
   const userTableProps = {
-    title: props.title,
+    title: title,
     action: {
       fn: addUser,
       actionText: 'Select user',
       actionIcon: <AddBox data-cy="select-user" />,
     },
-    selectedUsers: props.selectedUsers,
-    userRole: props.userRole || ('' as UserRole),
+    selectedUsers: selectedUsers,
+    userRole: userRole || ('' as UserRole),
     emailInvite: true,
-    selection: !!props.selection,
-    onUpdate: (data: BasicUserDetails[]) => props.addParticipants(data),
-    invitationUserRole: props.invitationUserRole || props.userRole,
+    selection: !!selection,
+    onUpdate: (data: BasicUserDetails[]) => addParticipants(data),
+    invitationUserRole: invitationUserRole || userRole,
+    selectedParticipants,
+    setSelectedParticipants,
   };
-
-  const isUserOfficer = useCheckAccess([UserRole.USER_OFFICER]);
-
   const peopleTable = <PeopleTable {...userTableProps} />;
-
   const proposalPeopleTable = <ProposalPeopleTable {...userTableProps} />;
 
   return (
     <Dialog
       aria-labelledby="simple-modal-title"
       aria-describedby="simple-modal-description"
-      open={props.show}
+      open={show}
       onClose={(_, reason) => {
         if (reason && reason == 'backdropClick') return;
-        props.close();
+        close();
       }}
       maxWidth="sm"
       fullWidth
     >
-      <DialogContent>
-        <IconButton
-          data-cy="close-modal-btn"
-          className={classes.closeButton}
-          onClick={() => props.close()}
-        >
+      <DialogTitle className={classes.dialogTitle}>
+        <IconButton data-cy="close-modal-btn" onClick={close}>
           <CloseIcon />
         </IconButton>
-        {props.participant && !isUserOfficer
-          ? proposalPeopleTable
-          : peopleTable}
+      </DialogTitle>
+      <DialogContent>
+        {participant && !isUserOfficer ? proposalPeopleTable : peopleTable}
       </DialogContent>
+      {selection && (
+        <DialogActions>
+          <div className={classes.selectedUsersText}>
+            {selectedParticipants.length} user(s) selected
```

Comment:
I think it shouldn't be possible because we set it initially to an empty array with this line:
`const [selectedParticipants, setSelectedParticipants] = useState<BasicUserDetails[]>([]);`
And after when we populate it we just push into that array or if we remove all the elements we set it to an empty array again.